scripts/cromwell/analyze_resource_acquisition.py

```diff
@@ -0,0 +1,460 @@
+	"""
+	if isinstance(m, list):
+		call_metadata = []
+		for m_shard in m:
+			call_metadata.extend(getCalls(m_shard, alias=alias))
+		return call_metadata
+
+	if 'labels' in m:
+		# In alias, track hierarchy of workflow/task up to current task nicely without repetition
+		if alias is None:
+			alias = ""
+		to_add = ""
+		if 'wdl-call-alias' in m['labels']:
+			to_add = m['labels']['wdl-call-alias']
+		elif 'wdl-task-name' in m['labels']:
+			to_add = m['labels']['wdl-task-name']
+		if to_add != "" and not alias.endswith(to_add):
+			if alias != "" and alias[-1] != ".":
+				alias += "."
+			alias += to_add
+
+	
+
+	call_metadata = []
+	if 'calls' in m:
+		for call in m['calls']:
+			# Skips scatters that don't contain calls
+			if '.' not in call:
+				continue
+			# In call_alias, track hierarchy of workflow/task up to current call nicely without repetition
+			if alias is None:
+				alias = ""
+			call_split = call.split('.')
+			call_name = call
+			if alias.endswith(call_split[0]):
+				call_name = call_split[1]
+			call_alias = alias
+			if call_alias != "" and call_alias[-1] != ".":
+				call_alias += "."
+			call_alias += call_name
+			# recursively get metadata
+			call_metadata.extend(getCalls(m['calls'][call], alias=call_alias))
+
+	if 'subWorkflowMetadata' in m:
+		call_metadata.extend(getCalls(m['subWorkflowMetadata'], alias=alias))
+
+	# in a call
+	if alias and ('monitoringLog' in m):
+		shard_index = '-2'
+		if 'shardIndex' in m:
+			shard_index = m['shardIndex']
+
+		attempt = '0'
+		if 'attempt' in m:
+			attempt = m['attempt']
+
+		job_id = 'na'
+		if 'jobId' in m:
+			job_id = m['jobId'].split('/')[-1]
+
+		start, end = calculate_start_end(m)
+
+		cpu, memory = get_mem_cpu(m)
+
+		cached = used_cached_results(m)
+		
+
+		callroot = 'na'
+		if 'callRoot' in m:
+			callroot = m['callRoot']
+
+		preemptible = was_preemptible_vm(m)
+		preemptible_cpu = 0
+		nonpreemptible_cpu = 0
+		if preemptible:
+			preemptible_cpu = cpu
+		else:
+			nonpreemptible_cpu = cpu
+			
+
+		hdd_size, ssd_size = get_disk_info(m)
+
+		call_metadata.append((start, 1, cpu, preemptible_cpu, nonpreemptible_cpu, memory, hdd_size, ssd_size))
+		call_metadata.append((end, -1, -1*cpu, -1*preemptible_cpu, -1*nonpreemptible_cpu, -1*memory, -1*hdd_size, -1*ssd_size))
+		if not preemptible:
+			global NUM_NONPREEMPTIBLE
+			global NONPREEMBTIBLE_TASKS
+			NUM_NONPREEMPTIBLE += 1
+			if alias in NONPREEMBTIBLE_TASKS:
+				NONPREEMBTIBLE_TASKS[alias] += 1
+			else:
+				NONPREEMBTIBLE_TASKS[alias] = 1
+
+		if cached:
+			global CACHED
+			global NUM_CACHED
+			NUM_CACHED += 1
+			if alias in CACHED:
+				CACHED[alias] += 1
+			else:
+				CACHED[alias] = 1
+
+	return call_metadata
+
+def check_workflow_valid(metadata, metadata_file, override_warning):
+	# these errors cannot be overcome
+	if 'status' not in metadata:
+		print("ERROR: Incomplete metadata input file %s. File lacks workflow status field." % metadata_file, file=sys.stderr)
+		exit(1)
```

Comment:
Throughout: I would `raise` an exception instead (see https://docs.python.org/3/library/exceptions.html). The choice of exception is up to you, here you could use `IOError` or just `RuntimeError`.

Reply:
Done